control-plane/grpc/src/pool/traits.rs

```diff
@@ -0,0 +1,295 @@
+use crate::{
+    grpc_opts::Context,
+    pool_grpc,
+    pool_grpc::{get_pools_request, CreatePoolRequest, DestroyPoolRequest},
+};
+use common_lib::{
+    mbus_api::{v0::Pools, ReplyError},
+    types::v0::{
+        message_bus::{CreatePool, DestroyPool, Filter, Pool, PoolState, PoolStatus},
+        store::pool::{PoolSpec, PoolSpecStatus},
+    },
+};
+use std::collections::HashMap;
+
+/// Trait implemented by services which support pool operations.
+/// This trait can only be implemented on types which support the PoolInfo trait.
+#[tonic::async_trait]
+pub trait PoolOperations {
+    async fn create(
+        &self,
+        pool: &(dyn CreatePoolInfo + Sync + Send),
```

Comment:
Same as before, could make CreatePoolInfo et all Send + Sync to avoid adding it on the signatures.